src/account.ts

```diff
@@ -1,645 +1,733 @@
 import { promisify } from 'util'
+import Web3 from 'web3'
+import { TransactionReceipt } from 'web3/types'
+import EthereumPlugin from '.'
+import { DataHandler, MoneyHandler } from './types/plugin'
 import {
-  IClaim,
-  IChannel,
+  ClaimablePaymentChannel,
+  fetchChannel,
   generateChannelId,
-  generateTx,
-  isSettling,
-  fetchChannel
+  isDisputed,
+  isValidClaimSignature,
+  PaymentChannel,
+  prepareTransaction,
+  remainingInChannel,
+  SerializedClaim,
+  SIGNED_MESSAGE_PREFIX,
+  spentFromChannel,
+  updateChannel
 } from './utils/contract'
-import Queue from 'p-queue'
-import { EventEmitter2 } from 'eventemitter2'
+import ReducerQueue from './utils/queue'
 
-BigNumber.config({ EXPONENTIAL_AT: 1e+9 }) // Almost never use exponential notation
+// Almost never use exponential notation
+BigNumber.config({ EXPONENTIAL_AT: 1e9 })
 
-enum TaskPriority {
-  ClaimTx = 4,
-  ChannelWatcher = 3,
-  Incoming = 2,
-  Outgoing = 1
-}
-
-export enum Unit {
-  Eth = 18,
-  Gwei = 9,
-  Wei = 0
-}
+const delay = (timeout: number) => new Promise(r => setTimeout(r, timeout))
 
-export const convert = (num: BigNumber.Value, from: Unit, to: Unit): BigNumber =>
-  new BigNumber(num).shiftedBy(from - to)
-
-export const format = (num: BigNumber.Value, from: Unit) =>
-  convert(num, from, Unit.Eth) + ' eth'
-
-export const getSubprotocol = (message: BtpPacket, name: string) =>
+const getBtpSubprotocol = (message: BtpPacket, name: string) =>
   message.data.protocolData.find((p: BtpSubProtocol) => p.protocolName === name)
 
-export const requestId = async () =>
+export const generateBtpRequestId = async () =>
   (await promisify(randomBytes)(4)).readUInt32BE(0)
 
+export const format = (num: AssetUnit) => convert(num, eth()) + ' eth'
+
+export interface SerializedAccountData {
+  accountName: string
+  receivableBalance: string
+  payableBalance: string
+  payoutAmount: string
+  ethereumAddress?: string
+  incoming?: ClaimablePaymentChannel
+  outgoing?: PaymentChannel
+}
+
 export interface AccountData {
-  // Hash/account identifier in ILP address
+  /** Hash/account identifier in ILP address */
   accountName: string
-  // Net amount in gwei the counterparty owes the this instance, including secured paychan claims
-  // - Negative implies this instance owes the counterparty
-  balance: BigNumber
-  // Sum of all fulfilled packets owed to the counterparty that have yet to be paid out (always >= 0)
-  // - The balance limits when settlements happen, but this limits the settlement amount so we don't send
-  //   all the clients' money directly back to them!
+
+  /** Incoming amount owed to us by our peer for their packets we've forwarded */
+  receivableBalance: BigNumber
+
+  /** Outgoing amount owed by us to our peer for packets we've sent to them */
+  payableBalance: BigNumber
+
+  /**
+   * Amount of failed outgoing settlements that is owed to the peer, but not reflected
+   * in the payableBalance (e.g. due to sendMoney calls on client)
```

Comment:
I find this description confusing, don't know how to make it better though.

Reply:
Yeah :/ Hopefully it'll become simpler after the payable/receivable balance move to the connector.